### PR TITLE
feature/NJS-3

### PR DESCRIPTION
**Pull request by MaciekPon**
add function to validate email
add authorization user schema

### CODE REVIEW

File: app/models/AuthUserSchema/AuthUserSchema.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  login: {
+    type: String,
+    lowercase: true,
+    trim: true,
+    required: true,
+    unique: true,
+  },
+  email: {
+    type: String,
+    lowercase: true,
```
Comment (ruljin) [inline app/models/AuthUserSchema/AuthUserSchema.js:17]: Same here. Is there any reason for e-mail to be lowercase?

File: app/models/AuthUserSchema/AuthUserSchema.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
+import mongoose from "mongoose";
+import bcrypt from "bcrypt";
```
Comment (ruljin) [inline app/models/AuthUserSchema/AuthUserSchema.js:2]: Going through the comment in which I said that schema middleware should be used in endpoints, there is no reason and sense in importing and using `bcrypt` here

**Review by ruljin — CHANGES_REQUESTED** [review]
Changes requested

File: app/models/AuthUserSchema/AuthUserSchema.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    validate: [validateEmail, "Email nieprawidłowy"],
+  },
+  password: {
+    type: String,
+    required: true,
+    validate: [4, "Hasło powinno posiadać minimum 4 znaki"],
+  },
+});
+
+AuthUserSchema.pre(save, function (next) {
```
[the remark this answers is not in the document]
Comment (JustD1d1t) [reply app/models/AuthUserSchema/AuthUserSchema.js:30]: Middlewares work after save parameteres, when schema/model is ready to save, that's why we use ".pre".  Ofc i can be wrong, but i think that how it workis

File: app/models/AuthUserSchema/AuthUserSchema.js [hunk 4]
```diff
@@ -0,0 +1,44 @@
+import mongoose from "mongoose";
+import bcrypt from "bcrypt";
+import { validateEmail } from "../validators";
+
+const SALT_WORK_FACTOR = 10;
+
+export const AuthUserSchema = new Schema({
+  login: {
+    type: String,
+    lowercase: true,
```
[the remark this answers is not in the document]
Comment (JustD1d1t) [reply app/models/AuthUserSchema/AuthUserSchema.js:10]: I don't know how it works with login/username. Is there differnece between for example "User" and "user" ? I think we should lowercase all logins to search them in our database like we should lowercase all emails

File: app/models/AuthUserSchema/AuthUserSchema.js (hunk 1 above)
Comment (JustD1d1t) [reply app/models/AuthUserSchema/AuthUserSchema.js:17]: same up

File: app/models/AuthUserSchema/AuthUserSchema.js [hunk 5]
```diff
@@ -0,0 +1,44 @@
+import mongoose from "mongoose";
+import bcrypt from "bcrypt";
+import { validateEmail } from "../validators";
```
[the remark this answers is not in the document]
Comment (JustD1d1t) [reply app/models/AuthUserSchema/AuthUserSchema.js:3]: i don't know this library. I knew way to validate by regexp. I think it's good way to know this package

File: app/middleware/hashPassword.js [hunk 6]
```diff
@@ -0,0 +1,8 @@
+import bcrypt from "bvrypt";
```
Comment (ruljin) [inline app/middleware/hashPassword.js:1]: What is `bvrypt` package? Besides, for me the directory should be named `utils`

File: app/models/UserSchema/UserSchema.js [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+    minLength: [3, "At least 3 characters"],
+  },
+  city: {
+    type: String,
+    required: true,
+    minLength: [3, "At least 3 characters"],
+  },
+  phone: {
+    type: Number,
+    required: true,
```
Comment (ruljin) [inline app/models/UserSchema/UserSchema.js:23]: For me, as we use `e-mail` as a login element (so, it is required) we shouldn't require `phone number`

File: app/models/UserSchema/UserSchema.js [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+    minLength: [8, "At least 8 characters"],
+    maxlength: [10, "Maximum 10 characters"],
+  },
+  announcements: [
+    { type: mongoose.Types.ObjectId, required: true, ref: "Announcement" },
```
Comment (ruljin) [inline app/models/UserSchema/UserSchema.js:28]: When I log in to the app first time, I don't have any announcements. So, is it required?

**Review by ruljin — CHANGES_REQUESTED** [review]
Changes requested

**Pull request merged**